### ISSUE TITLE
Works also with the newest 100Ah 12V BT

### ISSUE THREAD

**Issue by distancerunner** [chadj/renogy-smart-battery#2]
I tried it today with the 
rbt100 lfp12 bt

Every parameter except Charge State was shown. So it works perfectly

Thanks :)

**Comment by chadj** [comment]
Just curious, was charge state throwing an error or a "-" or something else?

**Comment by distancerunner** [comment]
It was empty. I started to debug the code  Because it's something new to learn. I will take look at this issue.

**Comment by chadj** [comment]
Ok!  FYI, that field should only have a value if the battery is charging or discharging

**Comment by distancerunner** [comment]
Yes you are right. Now it is charging and there are informations shown. I added a info about degree centigrade in temperature. An made a dark mode for me. Maybe I got time to add some graphics to made some parameters more pretty. 🤣 ![Screenshot_2022-06-11-11-30-01-576_com android chrome](https://user-images.githubusercontent.com/18682740/173182173-852f7579-0dfc-41da-aeb1-814a777960d3.jpg)

**Comment by distancerunner** [comment]
I forked it and made some changes. I added an average calculation for the actual current. So smooth out all values depending on the current to avoid jumps in charging estimations. 
https://github.com/distancerunner/renogy-smart-battery

**Comment by chadj** [comment]
Love it!  I’m happy the code is readable.  I didn’t document it much, I had hoped I kept it brief enough to be easily understood.  

I’ll get the changes merged when I can!

**Comment by distancerunner** [comment]
Yes, the code was absolutely fine. I have much more typescript mess at work in my application. :D

**Comment by distancerunner** [comment]
Sorry, you have little bit more to merge :D. I added a on phone stored statistic for 30 days (like the time on my votronics mppt controller). Maybe it needs a little bit more debugging. I will see how it runs the next days.
![image](https://user-images.githubusercontent.com/18682740/173329102-f276689f-d756-424d-b6f4-9e633017686f.png)

**Comment by chadj** [comment]
I merged these changes in.  One small fix here d92a0473b951b6c9580ce2fd55cfc4d140f7e76d
